Zend/zend_compile.c

```diff
@@ -8668,6 +8701,33 @@ static void zend_compile_prop_decl(zend_ast *ast, zend_ast *type_ast, uint32_t f
 					"Static property %s::$%s cannot be readonly",
 					ZSTR_VAL(ce->name), ZSTR_VAL(name));
 			}
+		} else {
+			if (flags & ZEND_ACC_PUBLIC_SET) {
+				zend_error_noreturn(E_COMPILE_ERROR,
+					"Non-readonly property %s::$%s may not use the public(set) modifier",
+					ZSTR_VAL(ce->name), ZSTR_VAL(name));
+			}
```

Comment:
I do not quite understand this. Why is that an error?

Reply:
Like, we do allow `protected protected(set)` ... but specifically `public public(set)` is to be disallowed? (unless readonly)

Reply:
You're right,  this is a leftover.